mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala

```diff
@@ -275,18 +316,33 @@ class GBTClassificationModel private[ml](
   @Since("2.0.0")
   lazy val featureImportances: Vector = TreeEnsembleModel.featureImportances(trees, numFeatures)
 
+  private def margin(features: Vector): Double = {
+    val treePredictions = _trees.map(_.rootNode.predictImpl(features).prediction)
+    blas.ddot(numTrees, treePredictions, 1, _treeWeights, 1)
+  }
+
   /** (private[ml]) Convert to a model in the old API */
   private[ml] def toOld: OldGBTModel = {
     new OldGBTModel(OldAlgo.Classification, _trees.map(_.toOld), _treeWeights)
   }
 
+  /**
+   * Note: this is currently an optimization that should be removed when we have more loss
+   * functions available than only logistic.
+   */
+  private lazy val loss = getOldLossType
```

Comment:
removed lazy, removed comment.  I made it lazy so as to not do the lookup if it doesn't need to be done, but since that isn't actually expensive and that only seemed to confuse it's better to remove it.